libcodechecker/cmd/cmd_line_client.py

```diff
@@ -372,28 +378,62 @@ def handle_list_result_types(args):
         for name in args.names:
             items.append((name, run_info.get(name)))
 
-    results_collector = []
-    for name, run_info in items:
-        run_id, run_date = run_info
-        results = client.getRunResultTypes(run_id, filters)
-
-        if args.output_format == 'json':
-            for res in results:
-                res.severity =\
-                    shared.ttypes.Severity._VALUES_TO_NAMES[res.severity]
-            results_collector.append({name: results})
-        else:  # plaintext, csv
-            print("Run '" + name + "', executed at '" + run_date + "'")
-            rows = []
-            header = ['Checker', 'Severity', 'Count']
-            for res in results:
-                sev = shared.ttypes.Severity._VALUES_TO_NAMES[res.severity]
-                rows.append((res.checkerId, sev, str(res.count)))
-
-            print(twodim_to_str(args.output_format, header, rows))
+    run_ids = [item[1][0] for item in items]
+
+    all_checkers_report_filter = codeCheckerDBAccess.ttypes.ReportFilter_v2()
+    all_checkers = client.getCheckerCounts(run_ids, all_checkers_report_filter,
+                                           None)
+    all_checkers_dict = dict((res.name, res) for res in all_checkers)
+
+    unrev_checkers = getStatistics(client, run_ids, 'reviewStatus',
+                                   [shared.ttypes.ReviewStatus.UNREVIEWED])
+
+    confirmed_checkers = getStatistics(client, run_ids, 'reviewStatus',
+                                       [shared.ttypes.ReviewStatus.CONFIRMED])
+
+    false_checkers = getStatistics(client, run_ids, 'reviewStatus',
+                                   [shared.ttypes.ReviewStatus.FALSE_POSITIVE])
+
+    wontfix_checkers = getStatistics(client, run_ids, 'reviewStatus',
+                                     [shared.ttypes.ReviewStatus.WONT_FIX])
+
+    resolved_checkers = getStatistics(client, run_ids, 'detectionStatus',
+                                      [shared.ttypes.DetectionStatus.RESOLVED])
+
+    all_results = []
+    for key, checker_data in sorted(all_checkers_dict.items(),
+                                    key=lambda x: x[1].severity,
+                                    reverse=True):
+        all_results.append(dict(
+            checker=key,
+            severity=shared.ttypes.Severity._VALUES_TO_NAMES[
+                checker_data.severity],
+            reports=checker_data.count,
+            unreviewed=checkerCount(unrev_checkers, key),
+            confirmed=checkerCount(confirmed_checkers, key),
+            false_positive=checkerCount(false_checkers, key),
+            wont_fix=checkerCount(wontfix_checkers, key),
+            resolved=checkerCount(resolved_checkers, key),
+         ))
 
     if args.output_format == 'json':
-        print(CmdLineOutputEncoder().encode(results_collector))
+        print(CmdLineOutputEncoder().encode(all_results))
+    else:
+        header = ['Checker', 'Severity', 'Reports', 'Unreviewed', 'Confirmed',
```

Comment:
I think this table looks a bit strange, as the "detection status" is too far.

I believe the following format would be better for the eye:

 * Checker
 * Severity
 * **All** reports (let's emphasise this)
 * Resolved
 * Unreviewed
 * Confirmed
 * False positive
 * Won't fix